flo-runner/src/main/java/com/spotify/flo/context/ForkingEvalContext.java

```diff
@@ -0,0 +1,344 @@
+    super(delegate);
+  }
+
+  public static EvalContext composeWith(EvalContext baseContext) {
+    return new ForkingEvalContext(baseContext);
+  }
+
+  @Override
+  public <T> Value<T> value(Fn<T> value) {
+    return super.value(fork(value));
+  }
+
+  private <T> Fn<T> fork(Fn<T> value) {
+    return () -> {
+
+      final ExecutorService executor = Executors.newCachedThreadPool();
+      Path tempdir = null;
+
+      try {
+        final Path workdir;
+        try {
+          tempdir = Files.createTempDirectory("flo-fork");
+          workdir = Files.createDirectory(tempdir.resolve("workdir"));
+        } catch (IOException e) {
+          throw new RuntimeException(e);
+        }
+
+        final Path closureFile = tempdir.resolve("closure");
+        final Path resultFile = tempdir.resolve("result");
+        final Path errorFile = tempdir.resolve("error");
+
+        final String home = System.getProperty("java.home");
+        final String classPath = System.getProperty("java.class.path");
+        final Path java = Paths.get(home, "bin", "java").toAbsolutePath().normalize();
+
+        try {
+          PersistingContext.serialize(value, closureFile);
+        } catch (Exception e) {
+          throw new RuntimeException("Failed to serialize closure", e);
+        }
+
+        final ProcessBuilder processBuilder = new ProcessBuilder()
+            .command(
+                java.toString(),
+                "-cp", classPath,
+                Trampoline.class.getName(),
+                closureFile.toString(),
+                resultFile.toString(),
+                errorFile.toString())
+            .directory(workdir.toFile());
+
+        final Process process;
+        try {
+          process = processBuilder.start();
+        } catch (IOException e) {
+          throw new RuntimeException(e);
+        }
+
+        // TODO: Log output
+
+        // Plan A
+        // ======
+        // 1. Set up logging in the child process to emit structured json (with severity and timestamps etc) to a file
+        // 2. Make the parent tail the log file, parse the json messages, decorate with task metadata and re-emit the
+        //    log messages. Note that re-emitting the log messages through the parent process slf4j logger would involve
+        //    figuring out how to keep the original log message metadata (e.g. timestamp, logger, thread, etc). Maybe
+        //    using the LocationAwareLogger interface. Printing the log messages directly to stderr might be more
+        //    straightforward. flo does not pull in a logging implementation, we'd have to pull one in (e.g. logback)
+        //    and set it up to do structured logging in the child process. Alternatively roll our own.
+        // 3. Also tail child process std{out,err} for unstructured output and log each line using the parent process
+        //    logger, taking care to not choke on huge lines and non-text output.
+
+        // Plan B
```

Comment:
This seems less disruptive.

Reply:
Went with [Plan C](https://github.com/spotify/flo/pull/84/commits/0534809920c445a725428fe6d9275220e88e6fc8#diff-171892128c20990cf4ce11dd33d5ca06R140):

```
Keep it simple.
1. Add FLO_TASK_ID to child process environment.
2. Rely on user provided logger to pick up FLO_TASK_ID.
3. Parent process just copies std{err,out} line by line. The reason for manual line-by-line copying instead
   of using Redirect.INHERIT is to ensure that line contents are not interleaved.
```